rsync_to_rdisc.py

```diff
@@ -215,31 +214,48 @@ def rsync_server_remote(hpc_server, client, to_be_transferred, mount_path, run_f
     rsync_succes = True
 
     for run in to_be_transferred:
-        with open(settings.log_path, 'a', newline='\n') as log_file:
-            log_file_writer = writer(log_file, delimiter='\t')
+        with open(settings.log_path, "a", newline="\n") as log_file:
+            log_file_writer = writer(log_file, delimiter="\t")
             log_file_writer.writerows([["#########"], [f"Date: {date}"], [f"Run_folder: {run}"]])
 
         transfer_settings = to_be_transferred[run]
-        # settings per folder data type, such as remote input dir and local output dir, etc.
+        # Settings per folder data type, such as remote input dir and local output dir, etc.
         missing = check_if_file_missing(transfer_settings["files_required"], f"{transfer_settings['input']}/{run}", client)
 
         if missing:
             print(transfer_settings, rsync_succes, missing, run, run_file)
             rsync_succes = action_if_file_missing(transfer_settings, rsync_succes, missing, run, run_file)
             continue  # don't transfer the run if a required file is missing.
-        os.system((
-            "rsync -rahuL --stats {user}@{server}:{input}/{run} {mount_path}/{output}/ "
-            " 1>> {log_path} 2>> {errorlog_path} 2> {temp_error_path}").format(
-                user=settings.user,
-                server=hpc_server,
-                input=transfer_settings["input"],
-                run=run,
-                mount_path=mount_path,
-                output=transfer_settings["output"],
-                log_path=settings.log_path,
-                errorlog_path=settings.errorlog_path,
-                temp_error_path=settings.temp_error_path
-        ))
+
+        # Get include and exclude patterns as list to easily access key-value pair in unit test.
+        if transfer_settings.get("include", None):
+            include_patterns = [f"--include '{pattern}'" for pattern in transfer_settings["include"]]
+        else:
+            include_patterns = ""
+        if transfer_settings.get("exclude", None):
+            exclude_patterns = [f"--exclude '{pattern}'" for pattern in transfer_settings["exclude"]]
+        else:
+            exclude_patterns = ""
+
+        source_destination = f"{settings.user}@{hpc_server}:{transfer_settings['input']}/{run}"
+        target_path = f"{mount_path}/{transfer_settings['output']}"
+        parse_stdout = f"1>> {settings.log_path} 2>> {settings.errorlog_path} 2> {settings.temp_error_path}"
+
+        rsync_cmd = [
+            "rsync",
+            "-rahuL",
+            "--stats",
+            source_destination,
+            target_path,
+            parse_stdout
+        ]
+
+        # Using list slicing to extend rsync_cmd list with include / exclude patterns.
+        rsync_cmd[3:3] = exclude_patterns
+        # Add include patterns before exclude patterns.
+        rsync_cmd[3:3] = include_patterns
```

Comment:
Why would you use list slicing to incorporate the exclude and include patterns to the rsync_cmd? To me it seems to make more sense to add them to the rsync_cmd list constructor. 

Considering `exclude_patterns` may look something like this `exclude_pattern = ["--exclude pattern_A", "--exclude pattern_B"]` **or** `exclude_pattern = []` (I changed the empty behaviour to [] in a previous comment). Then the variable can be added immediately to `rsync_cmd`

```suggestion
        rsync_cmd = [
            "rsync",
            "-rahuL",
            "--stats",
            *exclude_pattern, # add the values in the exclude_pattern list, to the rsync_cmd list, or add nothing if exclude_pattern is an empty list
            *include_pattern,
            source_destination,
            target_path,
            parse_stdout
        ]
```

Reply:
Nice!